### PR TITLE
Make HomeRecyclerItem an Entity

### PR DESCRIPTION
**Pull request by Siram-Ramesh**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -54,5 +59,17 @@ dependencies {
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'androidx.test.ext:junit:1.1.1'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
-
+    implementation "androidx.room:room-runtime:$rootProject.roomVersion"
+    kapt "androidx.room:room-compiler:$rootProject.roomVersion"
+    implementation "androidx.room:room-ktx:$rootProject.roomVersion"
+    androidTestImplementation "androidx.room:room-testing:$rootProject.roomVersion"
```
Comment (c10r) [inline app/build.gradle]: you're not actually using any android tests, so you can get rid of this

File: app/build.gradle [hunk 2]
```diff
@@ -54,5 +59,17 @@ dependencies {
...
+    implementation "androidx.lifecycle:lifecycle-extensions:$rootProject.archLifecycleVersion"
+    kapt "androidx.lifecycle:lifecycle-compiler:$rootProject.archLifecycleVersion"
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$rootProject.archLifecycleVersion"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    api "org.jetbrains.kotlinx:kotlinx-coroutines-core:$rootProject.coroutines"
+    api "org.jetbrains.kotlinx:kotlinx-coroutines-android:$rootProject.coroutines"
```
Comment (c10r) [inline app/build.gradle]: why are we adding these in this patch? why are they needed?
also, we should never need to use `api` in this project - `implementation` should suffice.

Read up on the difference [here](https://medium.com/mindorks/implementation-vs-api-in-gradle-3-0-494c817a6fa)

File: app/build.gradle [hunk 3]
```diff
@@ -51,8 +56,8 @@ dependencies {
     implementation 'androidx.lifecycle:lifecycle-extensions:2.2.0'
     implementation 'androidx.navigation:navigation-fragment-ktx:2.3.0'
     implementation 'androidx.navigation:navigation-ui-ktx:2.3.0'
-    testImplementation 'junit:junit:4.12'
```
Comment (c10r) [inline app/build.gradle:54]: let's remove ExampleUnitTest and ExampleInstrumentedTest as well since we're removing these

File: app/build.gradle [hunk 4]
```diff
@@ -16,6 +17,10 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 
+    packagingOptions {
```
Comment (c10r) [inline app/build.gradle]: runs fine on my laptop without this - why is this necessary?

**Review by c10r — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 4 above)
Comment (Siram-Ramesh) [reply app/build.gradle]: The reasoning the website gave for it was "Add the packagingOptions block inside the android block to exclude the atomic functions module from the package and prevent warnings."

Comment (c10r) [reply app/build.gradle]: Hmm what warnings do you see though? When I remove this I don't see any actual warnings

Comment (Siram-Ramesh) [reply app/build.gradle]: I wasn't seeing any, I guess I'll just remove it

**Review by c10r — APPROVED** [review]
